### ISSUE TITLE
TensorFlow iOS apps crash at launch on iPhone 4s/iPhone 5 simulators

### ISSUE THREAD

**Issue by timanglade** [tensorflow/tensorflow#8914]
Running [the simple iOS example](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/ios_examples/simple) provided by this repository will crash if I target any iPhone 4s/iPhone 5 **simulator**. The app will crash immediately at launch, before entering `didFinishLaunchingWithOptions`. The last call I can identify in the debugger is usually `tensorflow::internal::LogMessageFatal::~LogMessageFatal:`.

The crashes happen on iPhone 5 simulators running iOS 10.2, 9.3, and 9.1. And on iPhone 4s simulators running iOS 9.3, 9.1, and 8.4.

I can reproduce this behavior with other TensorFlow iOS apps.

The same apps that crash in an iPhone 4s/iPhone 5 simulator, will build & run fine on other simulators & devices, _e.g._ on iPad Air simulator running 10.2, a physical iPad Air running 10.2, or physical iPhone 6s Plus running 10.2. Tests on a device farm including iPhone 4s & iPhone 5 devices were inconclusive… I’m seeing a very similar crash happen, [mostly affecting devices running 9.3.5](https://docs.google.com/spreadsheets/d/1a1Erp-KPNA5bZuin5Fokzx0rVoqr2EEFEbKSEjEVpTw/edit), but the devices/iOS combos that fail on physical devices are slightly inconsistent from run to run (in the simulator the same devices crash 100% of the time no matter what OS I use, and from run to run).

### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
This [issue](https://github.com/tensorflow/tensorflow/issues/4640 ) is the only hit I get, but seems unrelated.

### Environment info
* Operating System: macOS Sierra 10.12.13
* Commit hash: b393fd79fd43242ea9be3ff3141f1809c412fd71 (master as of this writing)
* Bazel Version: 
    ```
    Build label: 0.4.3-jdk7
    Build target: bazel-out/local-    fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
    Build time: Thu Dec 22 12:31:38 2016 (1482409898)
    Build timestamp: 1482409898
    Build timestamp as int: 1482409898
    ```

### Minimal reproducible example

1. Prepare the simple iOS example provided by Tensorflow
    ```bash
    $ git clone https://github.com/tensorflow/tensorflow.git
    $ cd tensorflow
    $ tensorflow/contrib/makefile/download_dependencies.sh
    $ tensorflow/contrib/makefile/build_all_ios.sh
    $ open tensorflow/contrib/ios_examples/simple/tf_ios_makefile_example.xcodeproj
    ```
2. Download the Inception graph per the [instructions](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#ios)
3. Target the iPhone 5 simulator on iOS 10.2, and Run.
4. The app will crash before entering `didFinishLaunchingWithOptions`.
5. (If you target the iPad Air simulator on iOS 10.2, it will work.)

**Comment by petewarden** [comment]
I've reproduced this in the iPhone 5 simulator, and the console message is:
```
2017-04-03 08:29:17.629562: F tensorflow/core/platform/cpu_feature_guard.cc:35] The TensorFlow library was compiled to use SSE instructions, but these aren't available on your machine.
```
As a quick workaround, you can try changing the LOG(FATAL) in cpu_feature_guard.cc to a LOG(WARNING):
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/cpu_feature_guard.cc#L30
I believe this should be fixable by updating the build flags for i386, so I'll look into that. I fear this may be a different issue than the one that's causing intermittent crashes though, since this problem would always occur.

**Comment by timanglade** [comment]
> I fear this may be a different issue than the one that's causing intermittent crashes though, since this problem would always occur.

Yes, and good news on the intermittent crashes / physical devices front: I just tried doing test on a completely different device farm, and I’m not seeing any of these crashes. Sounds like it was an issue with the first device farm I was using…

My bad for linking the two behaviors: they had very similar behavior/symptoms, and I only discovered the Simulator crash while pursuing the device crash, but my hunch was probably incorrect! Thanks for your quick response @petewarden, sorry for the trouble.

**Comment by petewarden** [comment]
No problem at all, it was a good catch, and I'm very glad to hear your physical device issues are looking less problematic (knock on wood). I have PR #8918 pending for the simulator bug.

**Comment by petewarden** [comment]
I think this is actually related to #9809. In that bug I said we don't have a supported 32-bit x86 platform, but that's not true for the older iPhone emulators. I will close that one as a duplicate of this, and we'll aim to get this one fixed.

**Comment by petewarden** [comment]
This should be fixed now that #8918 is checked in.
